app/models/comfy/cms/page.rb

```diff
@@ -82,4 +82,8 @@ def suppress_mirrors_from_links_recirculation
   def ever_been_published?
     revisions.map {|revision| revision.data[:event]}.include?('published')
   end
+
+  def publishable?
+    keywords.map{ |tag| tag.value }.exclude?('do-not-publish')
```

Comment:
For bonus points you could push that check into `Tag` behind a predicate `publishable?` (although you'd need to flip the logic from `none?` to `any?` of course. If it has real meaning, it probably belongs over there in `Tag`.